pkg/storage/storage.go

```diff
@@ -483,39 +586,17 @@ func (s *storageModule) createSubvol(size uint64, name string, poolType pkg.Devi
 			continue
 		}
 
-		candidates = append(candidates, Candidate{
+		candidates = append(candidates, candidate{
 			Pool:      pool,
 			Available: usage.Size - (reserved + size), // available after new subvolume
 		})
 
-	}
-
-	if len(candidates) == 0 {
-		return nil, pkg.ErrNotEnoughSpace{DeviceType: poolType}
-	}
-
-	sort.Slice(candidates, func(i, j int) bool {
-		// reverse sorting so most available is at beginning
-		return candidates[i].Available > candidates[j].Available
-	})
-
-	var volume filesystem.Volume
-	for _, candidate := range candidates {
-		volume, err = candidate.Pool.AddVolume(name)
-		if err != nil {
-			log.Error().Err(err).Str("pool", candidate.Pool.Name()).Msg("failed to create new filesystem")
-			continue
-		}
-		if err = volume.Limit(size); err != nil {
-			candidate.Pool.RemoveVolume(volume.Name()) // try to recover
-			log.Error().Err(err).Str("volume", volume.Path()).Msg("failed to set volume size limit")
-			continue
+		// if we are looking for not mounted pools, break here
+		if !mounted {
```

Comment:
What's the point of this early return ?
Also what happens whit the previously unmounted pool that got mounted here, but had a too small available space ? Shouldn't they be unmounted and shutdown again ?

Reply:
Yes, good point. The early return was to make sure as we found the first candidate it would be immediatly returned.